go/tools/gazelle/packages/walk.go

```diff
@@ -52,3 +66,94 @@ func Walk(bctx build.Context, root string, f WalkFunc) error {
 		return f(pkg)
 	})
 }
+
+func findPackage(bctx build.Context, dir string) (*build.Package, error) {
+	packageGoFiles, otherFiles, err := findPackageFiles(dir)
+	if err != nil {
+		return nil, err
+	}
+
+	var packageNames []string
+	for name, _ := range packageGoFiles {
+		packageNames = append(packageNames, name)
+	}
+	packageName, err := selectPackageName(packageNames, dir)
+	if err != nil {
+		return nil, err
+	}
+
+	var files []os.FileInfo
+	files = append(files, packageGoFiles[packageName]...)
+	files = append(files, otherFiles...)
+	sort.Slice(files, func(i, j int) bool {
+		return files[i].Name() < files[j].Name()
+	})
+	bctx.ReadDir = func(dir string) ([]os.FileInfo, error) {
+		return files, nil
+	}
+	return bctx.ImportDir(dir, build.ImportComment)
+}
+
+func findPackageFiles(dir string) (packageGoFiles map[string][]os.FileInfo, otherFiles []os.FileInfo, err error) {
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return
+	}
+
+	packageGoFiles = make(map[string][]os.FileInfo)
+	for _, file := range files {
+		if file.IsDir() {
+			continue
+		}
+
+		name := file.Name()
+		filename := filepath.Join(dir, name)
+		ext := path.Ext(name)
+		isGo := ext == ".go"
+
+		var packageName string
+		if isGo {
+			fset := token.NewFileSet()
+			ast, err := parser.ParseFile(fset, filename, nil, parser.PackageClauseOnly)
+			if err != nil {
+				log.Printf("%s: error parsing package clause: %v", filename, err)
+				continue
+			}
+
+			packageName = ast.Name.Name
+			if strings.HasSuffix(packageName, "_test") {
+				packageName = packageName[:len(packageName)-len("_test")]
+			}
+			packageGoFiles[packageName] = append(packageGoFiles[packageName], file)
+		} else {
```

Comment:
invert this and continue
if !isGo {
  otherFiles = append(otherFiles, file)
  continue
}